AIPscan/Data/tests/test_storage_locations.py

```diff
@@ -179,3 +182,103 @@ def test_storage_locations_data(
         assert second_location[fields.FIELD_AIPS] == location_1_aips_count
         assert second_location[fields.FIELD_SIZE] == location_1_total_size
         assert second_location[fields.FIELD_FILE_COUNT] == location_1_file_count
+
+
+@pytest.mark.parametrize(
+    "start_date, end_date, expected_output",
+    [
+        # Test range of a few months.
+        ("2022-01-01", "2022-03-15", ["2022-01", "2022-02", "2022-03"]),
+        # Test a single month.
+        ("2022-03-12", "2022-03-14", ["2022-03"]),
+        # Test what happens if end is before start.
+        ("2022-06-01", "2022-05-01", []),
+        # Test that datetime.min is replaced by earliest AIP created date
+        (datetime.min, "2020-02-14", ["2020-01", "2020-02"]),
+        # Test that datetime.max is replaced by datetime.now()
+        (
+            datetime.now().isoformat()[:10],
+            datetime.max,
+            [datetime.now().isoformat()[:7]],
+        ),
+    ],
+)
+def test_get_months_covered_by_date_range(
+    storage_locations, start_date, end_date, expected_output
+):
+    start_date = parse_datetime_bound(start_date)
+    end_date = parse_datetime_bound(end_date, upper=True)
+    assert (
+        report_data._get_months_covered_by_date_range(1, start_date, end_date)
+        == expected_output
+    )
+
+
+@pytest.mark.parametrize(
+    "cumulative",
+    [
+        # Monthly totals
+        (False),
+        # Cumulative totals
+        (True),
+    ],
+)
+def test_storage_locations_by_month(storage_locations, cumulative):
+    """Test response from report_data.storage_locations_by_month endpoint."""
+    report = report_data.storage_locations_by_month(
+        storage_service_id=1,
+        start_date=parse_datetime_bound(DATE_BEFORE_AIP_1),
+        end_date=parse_datetime_bound(DATE_AFTER_AIP_3, upper=True),
+        cumulative=cumulative,
+    )
+
+    assert report[fields.FIELD_STORAGE_NAME] == "test storage service"
+
+    locations = report[fields.FIELD_LOCATIONS_BY_MONTH]
+
+    NUMBER_OF_MONTHS_IN_RANGE = 30
```

Comment:
I'm not sure why a variable assignment is necessary here?

Reply:
hm, good point. maybe I was anticipating re-using this variable but never did. I'll get rid of it.